azurelinuxagent/common/protocol/wire.py

```diff
@@ -804,9 +810,7 @@ def update_host_plugin():
                     return
 
                 def incarnation_changed():
-                    last_incarnation = None
-                    if os.path.isfile(incarnation_file):
-                        last_incarnation = fileutil.read_file(incarnation_file)
+                    last_incarnation = self.get_goal_state().incarnation
```

Comment:
should we do this until we remove the cache on disk? the current code always reads the file; get_goal_state may return the in-memory value

Reply:
I think we should keep this change on how we check for last_incarnation because without this the objects would again go out of sync as soon as some other thread updates the goal_state.
I can change the get_goal_state() to self.goal_state until the file cache is removed though.